### ISSUE TITLE
Latest htop src now compiles almost out-of-the-box on Mac - re: htop-osx.rb

### ISSUE THREAD

**Issue by edrozenberg** [Homebrew/homebrew#45197]
Regarding htop-osx.rb

The htop author has put work into making htop work on the Mac. I compiled the latest github htop source and it worked fine, with a couple of fixes from https://github.com/hishamhm/htop/issues/292#issuecomment-149709144 -> all credit to @skabashnyuk for figuring this out.

Specifically:

Patched darwin/Platform.h to add "SignalsPanel.h" headers:

```
#include "Action.h"
+#include "SignalsPanel.h"
#include "BatteryMeter.h"
#include "DarwinProcess.h"
````

Install ncurses from homebrew/dupes to support unicode (otherwise can pass --disable-unicode to configure):

````
brew tap homebrew/dupes
brew install ncurses
````

And then the htop compile for the github source:

````
git clone https://github.com/hishamhm/htop.git
cd htop
./autogen.sh
./configure CFLAGS="-I/usr/local/opt/ncurses/include" LDFLAGS="-L/usr/local/opt/ncurses/lib" CPPFLAGS="-I/usr/local/opt/ncurses/include"
make
````

I'm not a formula expert but could take a try at doing an htop formula for this recent source code, would it then be a new formula htop.rb rather than the  htop-osx.rb which is very mac specific?

**Comment by dunn** [comment]
We don't allow formulae in core with mandatory (or even recommended) dependencies on taps, so `--disable-unicode` will have to be the default behavior.

> would it then be a new formula htop.rb rather than the htop-osx.rb which is very mac specific?

Yep.  You'd also want to remove the alias that currently points `htop` to `htop-osx`: https://github.com/Homebrew/homebrew/blob/master/Library/Aliases/htop

**Comment by StarDuster** [comment]
I tried this with ncurses installed and I get a error on 10.11:
```
make                  
make/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
gcc -DHAVE_CONFIG_H -I.  -DNDEBUG -I/usr/local/opt/ncurses/include -pedantic -Wall -Wextra -std=c99 -D_XOPEN_SOURCE_EXTENDED -DSYSCONFDIR=\"/usr/local/etc\" -I"./darwin"  -I/usr/local/opt/ncurses/include -MT htop-AvailableMetersPanel.o -MD -MP -MF .deps/htop-AvailableMetersPanel.Tpo -c -o htop-AvailableMetersPanel.o `test -f 'AvailableMetersPanel.c' || echo './'`AvailableMetersPanel.c
In file included from AvailableMetersPanel.c:14:
./darwin/Platform.h:19:8: error: unknown type name 'SignalItem'
extern SignalItem Platform_signals[];
       ^
1 error generated.
make[1]: *** [htop-AvailableMetersPanel.o] Error 1
make: *** [all] Error 2

```

**Comment by edrozenberg** [comment]
@StarDuster you forgot to patch darwin/Platform.h? It should look like this after you modify it:

````
#include "Action.h"
#include "SignalsPanel.h"
#include "CPUMeter.h"
#include "BatteryMeter.h"
#include "DarwinProcess.h"
````

**Comment by StarDuster** [comment]
@edrozenberg I'm sorry for my stupid mistake,it works now

**Comment by kaefer** [comment]
As the header files are autogenerated one should better patch darwin/Platform.c (the header definitions are in there). I already made a pullrequest to the htop repo with this fix.

**Comment by edrozenberg** [comment]
@kaefer thanks for following up on that, the htop developer recently fixed the issue via https://github.com/hishamhm/htop/commit/a7fcbba75a3250f007baa4075f93dc1ce4f46225 so patching Platform.c/.h is no longer necessary.

**Comment by mistydemeo** [comment]
I think it should be possible to get unicode support enabled without a duplicate ncurses. OS X's ncurses does have wide-character support, even though it doesn't install an ncursesw; I see that htop's configure.ac just checks to see the library name, rather than for function availability, so it's mistakenly rejecting OS X's ncurses.

**Comment by edrozenberg** [comment]
Thanks Misty, I created a new issue at the htop project based on your feedback. The Mac's ncurses is version 5.x, and homebrew ncurses is 6.x - I also mentioned this in the htop issue in case it's relevant.

**Comment by edrozenberg** [comment]
@mistydemeo htop developer fixed the ncurses issue via recent commit https://github.com/hishamhm/htop/commit/347141502bd35f81f33cda69cea350fb984e5098

htop now compiles out-of-the-box on Mac with:

````
./autogen.sh
./configure --enable-unicode
make
````

I still need to put together a brew formula for this, should be easy since it doesn't need to do anything special now.

**Comment by mikemcquaid** [comment]
@edrozenberg If you do: ask him for a new tag and then we can build a new release.

**Comment by edrozenberg** [comment]
Note regarding building htop with default ncurses (version 5) from Mac OS X vs. ncurses (version 6) from homebrew:
* Building htop with default Mac OS X ncurses, scrolling with mouse inside htop does not work properly and causes htop to quit
* Building with a homebrew-installed ncurses (using the configure line with CFLAGS and LDFLAGS as described at the top of this issue), scrolling works properly.

For reference see https://github.com/hishamhm/htop/issues/322

**Comment by edrozenberg** [comment]
@mikemcquaid Waiting on a release tag from htop, looks like will happen later this month (Jan 2016).

**Comment by mikemcquaid** [comment]
@edrozenberg Great!

**Comment by hishamhm** [comment]
@mikemcquaid @edrozenberg As promised in FOSDEM, here's the release of htop 2.0!

http://hisham.hm/htop/releases/2.0.0/

Thank you for your patience! :)

**Comment by mikemcquaid** [comment]
Thanks for letting us known @hishamhm!

**Comment by UniqMartin** [comment]
Closing in favor of #49057.
